src/solver_intf.ml

```diff
@@ -0,0 +1,13 @@
+type 'a solver_module = (module Encoding.Solver_intf.S with type t = 'a)
+
+type solver = S : 'a solver_module * 'a -> solver
+
+module Solver = Encoding.Solver.Batch (Encoding.Z3_mappings)
+
+let solver_mod : Solver.t solver_module = (module Solver)
+
+let fresh_solver () =
+  let module Mapping = Encoding.Z3_mappings.Fresh.Make () in
+  let module Batch = Encoding.Solver.Batch (Mapping) in
+  let solver = Batch.create ~logic:QF_BVFP () in
+  S ((module Batch), solver)
```

Comment:
I don't understand why this module is named `solver_intf` and has functions like this one. See [the intf trick](https://www.craigfe.io/posts/the-intf-trick/) for an explanation of the `_intf` idiom.

Reply:
I don't know much about OCaml idioms, so please let me know how you think it would be better organized.

Reply:
`Solver` would be a better name

Reply:
And the `module Solver = ...` could be `module Z3 = ...` for now ?